src/dymaptic.GeoBlazor.Core/Components/Renderers/ColorRamps/AlgorithmicColorRamp.cs

```diff
@@ -0,0 +1,55 @@
+﻿using dymaptic.GeoBlazor.Core.Extensions;
+using dymaptic.GeoBlazor.Core.Objects;
+using dymaptic.GeoBlazor.Core.Serialization;
+using Microsoft.AspNetCore.Components;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace dymaptic.GeoBlazor.Core.Components.Renderers.ColorRamps;
+
+/// <summary>
+///     Creates a color ramp for use in a raster renderer. The algorithmic color ramp is defined by specifying two colors and the
+///     algorithm used to traverse the intervening color spaces.
+///     <a target="_blank" href="https://developers.arcgis.com/javascript/latest/api-reference/esri-rest-support-AlgorithmicColorRamp.html">
+///         ArcGIS
+///         JS API
+///     </a>
+/// </summary>
+public class AlgorithmicColorRamp : ColorRamp
+{
+
+    /// <summary>
+    ///     A string value representing the color ramp type.
+    /// </summary>
+    [JsonPropertyName("type")]
+    public override ColorRampType ColorRampType => ColorRampType.Algorithmic;
+
+    /// <summary>
+    ///     The algorithm used to generate the colors between the fromColor and toColor.
+    /// </summary>
+
+    public Algorithm Algorithm { get; set; }
```

Comment:
To support inheriting from MapComponent
1. Make all public properties here `[Parameter]`
2. Add two constructors, the empty one, and one that sets all the properties. Use that one for C# construction.

Reply:
parameters and constructors are now included